core/network/src/protocol/sync.rs

```diff
@@ -328,6 +371,7 @@ impl<B: BlockT> ChainSync<B> {
 		}
 	}
 
+	/// TODO: Explain what this is and why it exists
```

Comment:
I sort of understand what this code does (implements either of the ancestor search strategies) but I'm not quite sure how to document this or why this code is what it is.

Reply:
As a general remark, I think we tend to split codes at boundaries where it doesn't necessary make sense just to respect the rule of "not more than 5 levels of indentation".

Reply:
See https://github.com/paritytech/substrate/issues/1831 and  https://github.com/paritytech/substrate/pull/1875
The idea is to use exponential backoff till the first mismatching block and then do a binary search from there.